crates/hir-ty/src/display.rs

```diff
@@ -147,7 +148,7 @@ pub trait HirDisplay {
             curr_size: 0,
             max_size: None,
             omit_verbose_types: false,
-            display_target: DisplayTarget::SourceCode { module_id },
+            display_target: DisplayTarget::SourceCode { module_id, allow_opaque: allows_opaque },
```

Comment:
We currently render opaque types for `DisplayTarget::SourceCode` unconditionally, but it should fail unless we're rendering it for function param type or return type. I've looked through every `HirDisplay::display_source_code()` call sites and passed the flag accordingly, which I'd like to get careful review.